sources/packages/backend/apps/api/src/route-controllers/assessment/assessment.controller.service.ts

```diff
@@ -256,33 +258,80 @@ export class AssessmentControllerService {
-   *    disbursementReceipt1bcsl: 1005,
-   *    disbursementReceipt1bcag: 1006,
-   *    disbursementReceipt1bgpd: 1007,
-   *    disbursementReceipt1sbsd: 1008,
-   *    disbursementReceipt2cslf: 1000,
-   *    disbursementReceipt2csgp: 10001,
-   *    disbursementReceipt2bcsl: 1005,
-   *    disbursementReceipt2bcag: 1006,
-   *    disbursementReceipt2bgpd: 1007,
-   *    disbursementReceipt2sbsd: 1008,
+   */
+  private async populateDisbursementFinalAwardsValues(
+    assessment: StudentAssessment,
+    options?: {
+      studentId?: number;
+      applicationId?: number;
+    },
+  ): Promise<DynamicAwardValue | undefined> {
+    const [firstDisbursement] = assessment.disbursementSchedules;
+    if (
+      firstDisbursement.disbursementScheduleStatus !==
+      DisbursementScheduleStatus.Sent
+    ) {
+      // If a e-Cert was never generated no additional processing is needed.
+      return undefined;
+    }
+    let finalAward: DynamicAwardValue = undefined;
+    if (assessment.offering.offeringIntensity === OfferingIntensity.fullTime) {
+      // Full-time receipts are expected to contains all information about awards disbursed to students
+      // but specific BC grants. Receipts should be used as a source of the information for full-time application.
+      const disbursementReceipts =
+        await this.disbursementReceiptService.getDisbursementReceiptByAssessment(
+          assessment.id,
+          options,
+        );
+      if (!disbursementReceipts.length) {
+        // If the receipts are not available no additional processing is needed.
+        return finalAward;
+      }
+      let index = 1;
+      for (const schedule of assessment.disbursementSchedules) {
+        const awards = await this.populateDisbursementReceiptAwardValues(
+          disbursementReceipts,
+          schedule,
+          `${DISBURSEMENT_RECEIPT_PREFIX}${index++}`,
+        );
+        finalAward = { ...finalAward, ...awards };
       }
+      return finalAward;
+    }
+    // Part-time receipts will not contains all the awards value hence the e-Cert effective
+    // values are used instead to provide the best information as possible.
+    let index = 1;
+    for (const schedule of assessment.disbursementSchedules) {
+      const awards = this.populateDisbursementECertAwardValues(
+        schedule.disbursementValues,
+        `${DISBURSEMENT_RECEIPT_PREFIX}${index++}`,
+      );
+      finalAward = { ...finalAward, ...awards };
+    }
+    return finalAward;
+  }
+
+  /**
+   * Populate final awards values from disbursements receipts.
+   * @param disbursementReceipts disbursement receipts.
+   * @param disbursementSchedule disbursement schedules.
+   * @param identifier identifier which is used to create dynamic data by appending award code to it.
+   * @returns dynamic award data of disbursement receipts of a given disbursement.
    */
-  private populateDisbursementReceiptAwardValues(
+  private async populateDisbursementReceiptAwardValues(
```

Comment:
Why this method is async?

Reply:
Back and forth between refactors, I will remove it.